### ISSUE TITLE
[padding-line-between-statements] Wrong blank line for property called "type"

### ISSUE THREAD

**Issue by madcampos** [typescript-eslint/typescript-eslint#4040]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/padding-line-between-statements": [
	"warn",
	{ "blankLine": "always", "prev": "*", "next": "type" },
	{ "blankLine": "always", "prev": "type", "next": "*" }
      ],
  }
}
```

```TS
let var1, var2, type;

var1 = 'bar';
type = 'baz';
var2 = 'quux';
```

**Expected Result**

It should differentiate between a `type` declaration and a variable named `type`.

```TS
let var1, var2, type;

var1 = 'bar';
type = 'baz';
var2 = 'quux';
```

**Actual Result**

```TS
let var1, var2, type;

var1 = 'bar';

type = 'baz';

var2 = 'quux';
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.1.0` |
| `@typescript-eslint/parser`        | `5.1.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.1` |
| `node`                             | `16.11.1` |
